Quadriga/src/main/java/edu/asu/spring/quadriga/accesschecks/impl/ProjectSecurityChecker.java

```diff
@@ -190,34 +190,33 @@ public boolean checkProjectAccess(String userName, String projectId)
         chkAccess = this.isProjectOwner(userName, projectId);
 
         if (!chkAccess) {
-            chkAccess = this.isUserCollaboratorOnProject(userName, projectId,
-                    RoleNames.ROLE_QUADRIGA_ADMIN);
+            chkAccess = this.isUserCollaboratorOnProject(userName, projectId, RoleNames.ROLE_QUADRIGA_ADMIN);
         }
         return chkAccess;
     }
 
-    /**
-     * This method checks if the user has the editor role to
-     * this Project
-     * 
-     * @param userName
-     * @return boolean - TRUE if the user is editor on the project; otherwise false
-     * @throws QuadrigaStorageException
-     * @author kiranbatna
-     */
     @Override
     @Transactional
-    public boolean isEditor(String userName,
-            String projectId) throws QuadrigaStorageException {
-        // check if the user is project editor
-        return accessManager
-                .isUserEditorOfProject(userName, projectId) > 0;
+    public boolean isUnixnameInUse(String unixName, String projectId) throws QuadrigaStorageException {
+        return accessManager.getProjectIdByUnixName(unixName) != null;
     }
 
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.spring.quadriga.accesschecks.IProjectSecurityChecker#
+     * getCollaboratorRoles(java.lang.String, java.lang.String)
+     */
     @Override
-    @Transactional
-    public boolean isUnixnameInUse(String unixName, String projectId)
-            throws QuadrigaStorageException {
-       return accessManager.getProjectIdByUnixName(unixName) != null;
+    public List<String> getCollaboratorRoles(String userName, String projectId) {
+        List<String> userDBRoles = accessManager.getProjectCollaboratorRoles(userName, projectId);
+        List<String> collaboratorRoles = new ArrayList<String>();
+
+        for (int i = 0; i < userDBRoles.size(); i++) {
+            collaboratorRoles.add(
+                    roleManager.getQuadrigaRoleByDbId(IQuadrigaRoleManager.PROJECT_ROLES, userDBRoles.get(i)).getId());
+        }
```

Comment:
let's be adventurous  and use some Java 8 lambda expression here ;)
(in other words, turn this into a one liner)